opentelemetry-exporter-cloud-trace/tests/test_gcp_resource_detector.py

```diff
@@ -46,11 +57,75 @@ def test_finding_gce_resources(self, getter):
         )
 
 
+def pop_environ_key(key):
+    if key in os.environ:
+        os.environ.pop(key)
+
+
+def clear_gke_env_vars():
+    pop_environ_key("CONTAINER_NAME")
+    pop_environ_key("NAMESPACE")
+    pop_environ_key("HOSTNAME")
+
+
+class TestGKEResourceFinder(unittest.TestCase):
+    def tearDown(self) -> None:
+        clear_gke_env_vars()
+
+    @mock.patch("opentelemetry.tools.resource_detector.requests.get")
+    def test_missing_environment_variables(self, getter):
+        getter.return_value.json.return_value = GKE_RESOURCES_JSON_STRING
+
+        # If one of CONTAINER_NAME, NAMESPACE is missing from the environment
+        # return no resources
+
+        pop_environ_key("CONTAINER_NAME")
+        pop_environ_key("NAMESPACE")
+        self.assertEqual(get_gke_resources(), {})
+
+        os.environ["NAMESPACE"] = "namespace"
+        self.assertEqual(get_gke_resources(), {})
+
+        pop_environ_key("NAMESPACE")
+        os.environ["CONTAINER_NAME"] = "container_name"
```

Comment:
Repeating a string like `"NAMESPACE"` or `"CONTAINER_NAME"` in code like this is vulnerable to typos; if it's mistyped in one spot it can break the test and be hard to debug.

A trick to avoid that is to define these strings in constants, then reuse that constant every time the magic string is needed. That will turn typos into compile errors (because the typo is in a variable name instead of a magic string) and make debugging those typos much easier.